worker/worker.go

```diff
@@ -422,6 +422,33 @@ func (w *worker) withRevision(ctx context.Context, contractID types.FileContract
 	return fn(revision)
 }
 
+func (w *worker) withRevisionV3(ctx context.Context, contractID types.FileContractID, hk types.PublicKey, siamuxAddr string, lockPriority int, lockDuration time.Duration, fn func(revision types.FileContractRevision) error) error {
+	// acquire contract lock
+	if lockID, err := w.bus.AcquireContract(ctx, contractID, lockPriority, lockDuration); err != nil {
+		return fmt.Errorf("%v: %w", "failed to acquire contract for funding EA", err)
+	} else {
+		defer func() {
+			if err := w.bus.ReleaseContract(ctx, contractID, lockID); err != nil {
+				w.logger.Errorw(fmt.Sprintf("failed to release contract, err: %v", err), "hk", hk, "fcid", contractID)
+			}
+		}()
+	}
+
+	// fetch contract revision
+	pt, valid := w.priceTables.PriceTable(hk)
+	var rev types.FileContractRevision
+	var err error
+	if valid {
+		rev, err = w.FetchRevisionWithContract(ctx, &pt, hk, siamuxAddr, contractID)
```

Comment:
Is there a particular reason for that? We could use EA payments regardless of the state of the pricetable.
The reason I've chosen contract payment for now is because we only use `withRevisionV3` for syncing and funding accounts which are probably the two things that have the highest chance of dealing with empty accounts and since they happen rarely anyway.